test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -391,6 +409,119 @@ public void updateRetentionPolicyForCBRTest() throws Exception {
+        // Read one event with reader.
+        readingEventsFromStream(1, reader);
+
+        checkpoint = initiateCheckPoint("Checkpoint3", readerGroup, reader, clock);
+        read = reader.readNextEvent(READ_TIMEOUT);
+        log.info("Reading next event after checkpoint3 {}", read.getEvent());
+        cpResult = checkpoint.join();
+        assertTrue(checkpoint.isDone());
+
+        //Controller Failover
+        scaleAndUpdateControllerURI(1);
+        log.info("Successfully scaled down controller to 1 instance");
+        //SegmentStore Failover
+        Futures.getAndHandleExceptions(segmentStoreService.scaleService(1), ExecutionException::new);
+        log.info("Successfully scaled down segment store to 1 instance");
+
+        // Retention set has two stream cut at 0/150...0/240
+        // READER_GROUP_1 updated stream cut at 0/90
+        // Subscriber lower bound is 0/90, truncation should happen at this point
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 90.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(scope, stream), 0L).join().values().stream().anyMatch(off -> off == 90),
+                5000,  2 * 60 * 1000L);
+
+        // Validating the restart scenario - Start
+        checkpoint = initiateCheckPoint("Checkpoint4", readerGroup, reader, clock);
+        read = reader.readNextEvent(READ_TIMEOUT);
+        log.info("Reading next event after checkpoint4 {}", read.getEvent());
+        cpResult = checkpoint.join();
+        assertTrue(checkpoint.isDone());
+
+        //Closing all the resources before restart
+        reader.close();
+        readerGroup.close();
+        readerGroupManager.close();
+        writer.close();
+        clientFactory.close();
+        connectionFactory.close();
+        Futures.getAndHandleExceptions(controllerService.scaleService(0), ExecutionException::new);
+        log.info("Successfully stopped 1 instance of controller service");
+        Futures.getAndHandleExceptions(segmentStoreService.scaleService(0), ExecutionException::new);
+        log.info("Successfully stopped 1 instance of segment store service");
+
+        Futures.getAndHandleExceptions(segmentStoreService.scaleService(1), ExecutionException::new);
+        log.info("Successfully started 1 instance of segment store service");
+        scaleAndUpdateControllerURI(1);
+        log.info("Successfully started 1 instance of controller service");
+
+        // Retention set has two stream cut at 0/150...0/240
+        // READER_GROUP_1 updated stream cut at 0/120
+        // Subscriber lower bound is 0/120, truncation should happen at this point
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 120.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(scope, stream), 0L).join().values().stream().anyMatch(off -> off == 120),
+                5000,  2 * 60 * 1000L);
```

Comment:
Here too

Reply:
Raised an issue to track this : https://github.com/pravega/pravega/issues/7037